scripts/pylib/twister/twisterlib.py

```diff
@@ -809,13 +809,19 @@ def run_custom_script(script, timeout):
     def handle(self):
         out_state = "failed"
         runner = None
+        timeout_cnt = self.timeout
 
         hardware = self.device_is_available(self.instance)
-        while not hardware:
+        while not hardware and timeout_cnt > 0:
             logger.debug("Waiting for device {} to become available".format(self.instance.platform.name))
             time.sleep(1)
+            timeout_cnt -= 1
             hardware = self.device_is_available(self.instance)
 
+        if not hardware:
+            self.instance.reason = "Timeout"
```

Comment:
I think we need to set state for instance and missing testcases too, because this will return immediately.

Reply:
Thanks I add set_state so that it will be marked as skipped, instead of running.
```
self.instance.status, _ = self.instance.handler.get_state()
```
BTW I guess this is missused, as the handler status shall not set to instance.status, which is not related.